Core/Core/Files/FileUploader.swift

```diff
@@ -0,0 +1,325 @@
+//
+// Copyright (C) 2019-present Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CoreData
+
+public protocol FileUploaderDelegate {
+    func fileUploader(_ fileUploader: FileUploader, finishedUploadingFile file: File, inContext context: NSManagedObjectContext)
+}
+
+public class FileUploader: NSObject, URLSessionDelegate, URLSessionTaskDelegate, URLSessionDataDelegate {
+    fileprivate struct Session: Codable {
+        let bundleID: String
+        let appGroup: AppGroup?
+        let userID: String
+        let baseURL: URL
+        let actAsUserID: String?
+
+        init(bundleID: String, appGroup: AppGroup?, userID: String, baseURL: URL, actAsUserID: String?) {
+            self.bundleID = bundleID
+            self.appGroup = appGroup
+            self.userID = userID
+            self.baseURL = baseURL
+            self.actAsUserID = actAsUserID
+        }
+
+        init?(string: String) {
```

Comment:
`init?(identifier: String)` to make a clear connection with the `identifier` below.

Reply:
Done